lib/commons/dom/url-props-from-attribute.js

```diff
@@ -0,0 +1,154 @@
+/* global dom */
+
+/**
+ * Parse resource object for a given node from a specified attribute
+ * @method urlPropsFromAttribute
+ * @param {HTMLElement} node given node
+ * @param {String} attribute attribute of the node from which resource should be parsed
+ * @returns {Object}
+ */
+dom.urlPropsFromAttribute = function urlPropsFromAttribute(node, attribute) {
+	const value = node[attribute];
+	if (!value) {
+		return undefined;
+	}
+
+	const nodeName = node.nodeName.toUpperCase();
+	let parser = node;
+
+	/**
+	 * Note:
+	 * The need to create a parser, is to keep this function generic, to be able to parse resource from element like `iframe` with `src` attribute
+	 */
+	if (!['A', 'AREA'].includes(nodeName)) {
+		parser = document.createElement('a');
+		parser.href = value;
+	}
+
+	const { pathname = '', filename = '' } = getPathnameOrFilename(
+		parser.pathname
+	);
+	return {
+		protocol: parser.protocol,
+		hostname: parser.hostname,
+		port: getPort(parser.port),
+		pathname: pathname.replace(/\/$/, ''), // remove trialing slashes
+		search: getSearchPairs(parser.search),
+		hash: getHashRoute(parser.hash),
+		filename
+	};
+};
+
+
+/**
+ * Resolve given port excluding default port(s)
+ * @param {String} port port
+ * @returns {String}
+ */
+function getPort(port) {
+	/**
+	 * Note: IE11 resolves port to `443` for any URL with `https:`
+	 */
+	const excludePorts = [`443`, `80`]
+	return !excludePorts.includes(port)
+		? port
+		: ``
+}
+
+/**
+ * Resolve if a given pathname has filename & resolve the same as parts
+ * @method getPathnameOrFilename
+ * @param {String} pathname pathname part of a given uri
+ * @returns {Array<String>}
+ */
+function getPathnameOrFilename(pathname) {
+	if (!pathname) {
+		return;
```

Comment:
This is going to throw. Because you're doing destructive assignment on the result, this must return an object. This code path isn't executed in your tests.